woodwork/table_accessor.py

```diff
@@ -73,81 +82,151 @@ def init(self,
+            column_origins (str, Dict[str -> str], optional): Origin of each column. If a string is supplied, it is
                 used as the origin for all columns. A dictionary can be used to set origins for individual columns.
+            validate (bool, optional): Whether parameter and data validation should occur. Defaults to True. Warning:
+                Should be set to False only when parameters and data are known to be valid.
+                Any errors resulting from skipping validation with invalid inputs may not be easily understood.
+        """
+        self.init_with_partial_schema(**kwargs)
+
+    def init_with_full_schema(self, schema: TableSchema, validate: bool = True, **kwargs) -> None:
+        """Initializes Woodwork typing information for a DataFrame with a complete schema.
+
+        Args:
+            schema (Woodwork.TableSchema): Typing information to use for the DataFrame instead of performing inference. Note that any changes made to the
+                schema object after initialization will propagate to the DataFrame. Similarly, to avoid unintended typing information changes, the same schema
+                object should not be shared between DataFrames.
+            validate (bool, optional): Whether parameter and data validation should occur. Defaults to True. Warning: Should be set to False only when
+                parameters and data are known to be valid. Any errors resulting from skipping validation with invalid inputs may not be easily understood.
+        """
+        if validate:
+            _check_schema(self._dataframe, schema)
+            _check_unique_column_names(self._dataframe)
+        self._schema = schema
+
+        extra_params = [key for key in kwargs]
+        if extra_params:
+            warnings.warn("A schema was provided and the following parameters were ignored: " + ", ".join(extra_params), ParametersIgnoredWarning)
+
+    def init_with_partial_schema(self,
+                                 schema: TableSchema = None,
+                                 index: str = None,
+                                 time_index: str = None,
+                                 logical_types: Dict[ColumnName, Union[str, LogicalType, None]] = None,
+                                 already_sorted: bool = False,
+                                 name: str = None,
+                                 semantic_tags: Dict[ColumnName, Union[str, List[str], Set[str]]] = None,
+                                 table_metadata: dict = None,
+                                 column_metadata: Dict[ColumnName, dict] = None,
+                                 use_standard_tags: Union[bool, UseStandardTagsDict] = None,
+                                 column_descriptions: Dict[ColumnName, str] = None,
+                                 column_origins: Union[str, Dict[ColumnName, str]] = None,
+                                 validate: bool = True,
+                                 **kwargs) -> None:
```

Comment:
After reading through @davesque's comments I think these changes should be possible to do to let us remove the `Dict` and `Union` imports (unless I'm misunderstanding where we can use `dict` vs `Dict`)

```suggestion
                                 schema: TableSchema = None,
                                 index: str = None,
                                 time_index: str = None,
                                 logical_types: dict[ColumnName, str | LogicalType | None] = None,
                                 already_sorted: bool = False,
                                 name: str = None,
                                 semantic_tags: dict[ColumnName, str | list[str] | set[str]] = None,
                                 table_metadata: dict = None,
                                 column_metadata: dict[ColumnName, dict] = None,
                                 use_standard_tags: bool | UseStandardTagsDict = None,
                                 column_descriptions: dict[ColumnName, str] = None,
                                 column_origins: str | dict[ColumnName, str] = None,
                                 validate: bool = True,
                                 **kwargs) -> None:
```

(can also make similar changes in `_infer_missing_logical_types` and `_merge_use_standard_tags` )

Reply:
@tamargrey Actually, @tuethan1999 and I realized that it might be a bit too early to adopt those next gen type annotation features.  I think `Union` and `Dict` still make sense at this point.  Otherwise, we might end up having to use python 3.9 to do the type checking with mypy.  That might be okay, but it could just as well introduce some complexities to our CI pipeline that didn't exist before.  It might also introduce subtle issues that come up as a result of the version mismatch.  Not sure how that would look but it seems possible.

Reply:
Ah, thanks for the explanation! I agree that being safe here makes sense!